src/main/java/Parser.java

```diff
@@ -0,0 +1,199 @@
+            } else if (commandArray[0].toLowerCase().equals("find")){
+                if (commandArray.length >= 2) {
+                    findTask(taskList, commandArray[1]);
+                } else {
+                    throw new DukeArgumentException("Please specify the keyword(s) that you want to search.");
+                }
+            } else if (commandArray[0].toLowerCase().equals("todo")) {
+                if (commandArray.length >= 2) {
+                    addTask(taskList, commandArray[1], "T");
+                } else {
+                    throw new DukeArgumentException("Missing field in todo command.");
+                }
+            } else if (commandArray[0].toLowerCase().equals("deadline")) {
+                if (commandArray.length >= 2) {
+                    addTask(taskList, commandArray[1], "D");
+                } else {
+                    throw new DukeArgumentException("Missing field in deadline command.");
+                }
+            } else if (commandArray[0].toLowerCase().equals("event")) {
+                if (commandArray.length >= 2) {
+                    addTask(taskList, commandArray[1], "E");
+                } else {
+                    throw new DukeArgumentException("Missing field in event command.");
+                }
+            } else {
+                throw new DukeUnknownException("Apologies, I do not recognise this command.");
+            }
+        } catch (DukeException err) {
+            Ui.printWithBorder(err.getMessage());
+        }
+    }
+
+    /**
+     * Adds Task to TaskList based on its type (i.e. Todo, Deadline or Event).
+     * @param taskList User's TaskList.
+     * @param input Input provided by the user.
+     * @param type Type of Task ("T" for Todo, "D" for Deadline, "E" for Event).
+     * @throws DukeTaskException
+     */
+    public static void addTask(TaskList taskList, String input, String type) throws DukeTaskException {
+        String str = "\nCurrent number of task(s): ";
+
+        if (type.equals("T")) {
+            Task todo = new Todo(input);
+            taskList.add(todo);
+            Ui.printWithBorder("The following to-do has been added:\n    "
+                    + todo.toString() + str + taskList.size());
+
+        } else if (type.equals("D")) {
+            String[] arr = input.split("/", 2);
+            if (arr.length > 1) {
+                String description = arr[0];
+                String by = arr[1].split(" ", 2)[1];
+                if (isLocalDate(by)) {
+                    LocalDate deadlineDate = LocalDate.parse(by);
+                    Deadline deadline = new Deadline(description, deadlineDate);
+                    taskList.add(deadline);
+                    Ui.printWithBorder("The following task has been added:\n" +
+                            "    " + deadline.toString() + str + taskList.size());
+                } else {
+                    throw new DukeTaskException("Invalid date format detected. " +
```

Comment:
The believe the + should be pushed to the start of the next line. (I had a style error for my own code when I ran the Gradle style check)

(e.g. 
stringA
+ stringB)